omnibus/files/private-chef-cookbooks/private-chef/attributes/default.rb

```diff
@@ -623,6 +623,7 @@
 default['private_chef']['postgresql']['archive_mode'] = "off" # "cannot be enabled when wal_level is set to minimal"
 default['private_chef']['postgresql']['archive_command'] = ""
 default['private_chef']['postgresql']['archive_timeout'] = 0 # 0 is disabled.
+default['private_chef']['postgresql']['sslmode'] = 'disable'
```

Comment:
It might be worth documenting that the allowable choices come from the PGSSLMODE environment var for libpq, and what they are.
It looks like we support disable, prefer, and require, but not allow?
I'm taking this info from https://www.postgresql.org/docs/9.6/libpq-connect.html sslmode.

Reply:
Inserted comment (below):

`default['private_chef']['postgresql']['archive_command'] = ""
default['private_chef']['postgresql']['archive_timeout'] = 0 # 0 is disabled.
# see: https://www.postgresql.org/docs/9.6/libpq-ssl.html
# allowable sslmodes are: disable | allow | prefer | require | verify-ca | verify-full
default['private_chef']['postgresql']['sslmode'] = 'disable'`

I'm assuming 'allow' would work, but at the moment 'verify-ca' and 'verify-full' have not been implemented chef-server side.  Could you give me more thoughts on 'allow'?

Reply:
Oh, I think you mean this:
```
<%= { 'disable' => false, 'prefer' => true, 'require' => 'required' }[@postgresql['sslmode']] %>
```

Reply:
Exactly; that bit of code limits which options we accept and is probably worth documenting above.

Reply:
So, if we want to add other ssl options, like `allow`...

Seems we use this as our driver: https://github.com/chef/epgsql-1 It supports `ssl = false | true | required` (which maps directly onto a subset of what postgresql supports), and `ssl_opts = ssl:ssl_option()`.  Well, good luck finding out what `ssl:ssl_option()` can be (somewhat of a rabbit hole).  I've looked in ssl app, ssl_api.hrl, and done various google searches.  I found an `ssloption` in http://erlang.org/documentation/doc-1/man/ssl.html and an `ssl_option` in https://erldocs.com/17.4.1/ssl/ssl_connection.html .  For now, near as i can tell the driver wants to shoehorn us into `ssl = false | true | required`.